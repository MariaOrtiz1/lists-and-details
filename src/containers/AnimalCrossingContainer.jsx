import React, { Component } from 'react';
import { findVillagers } from '../services/animalCrossing';
import VillagerList from '../components/villagers/VillagerList';


export default class AnimalCrossingContainer extends Component {
    state = {
      loading: true,
      villagers: [],
    };

    componentDidMount() {
      findVillagers().then((villagers) => 
        this.setState({ villagers, loading: false })
      );
    }

    render() {
      const { loading, villagers } = this.state;
      if(loading) {
        return (
          <img 
            src="https://i.giphy.com/media/XE7hLtlphwAK70mP3i/giphy.webp" 
            alt="loading spinner of flying package with balloon" 
            style={{
              'display': 'flex',
              'justifyContent': 'center',
              'alignSelf': 'center',
            }}/>
        );
      }

      return <VillagerList villagers={villagers} />;
    }
}
