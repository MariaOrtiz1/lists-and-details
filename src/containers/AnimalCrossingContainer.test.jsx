import React from 'react';
import { render, screen } from '@testing-library/react';
import AnimalCrossingContainer from './AnimalCrossingContainer';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import { MemoryRouter } from 'react-router-dom';
import listData from '../data/listData.json';

const mockServer = setupServer(
  rest.get('https://ac-vill.herokuapp.com/villagers/', (req, res, ctx) => {
    return res(ctx.json(listData));
  })
);

describe('AnimalCrossingContainer', () => {
  beforeAll(() => mockServer.listen());
  afterAll(() => mockServer.close());
  
  it('renders a list of villagers to the page', async () => {
    const { container } = render(
      <MemoryRouter>
        <AnimalCrossingContainer />
      </MemoryRouter>);
  
    screen.getByAltText('loading spinner of flying package with balloon');
  
    const ul = await screen.findByRole('list', { name: 'villagers' });
  
    expect(ul).not.toBeEmptyDOMElement();
    expect(container).toMatchSnapshot();
  });
});
