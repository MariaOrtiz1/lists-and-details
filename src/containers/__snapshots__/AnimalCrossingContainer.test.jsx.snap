// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimalCrossingContainer renders a list of villagers to the page 1`] = `
<ul
  aria-label="villagers"
  role="list"
>
  <li>
    <button
      style="display: flex; flex-direction: column; align-items: center; background-color: rgb(162, 124, 191); border-radius: 30px;"
    >
      <p />
      <p>
        Aurora
      </p>
      <img
        alt="Aurora"
        src="example.com/image.png"
        style="height: 300px; width: 250px; background-color: rgb(237, 202, 219); border-radius: 30px; border: 3px dotted #a27cbf;"
      />
      <p>
        penguin
      </p>
      <p>
        normal
      </p>
      <p>
        Always keep your cool.
      </p>
    </button>
  </li>
</ul>
`;
